packages/sit-onyx/src/components/OnyxDataGrid/OnyxDataGrid.vue

```diff
@@ -0,0 +1,44 @@
+<script setup lang="ts" generic="TEntry extends DataGridEntry">
+import { ref, toRefs, watch, type Ref } from "vue";
+import {
+  OnyxDataGridRenderer,
+  type DataGridEntry,
+  type DataGridMetadata,
+  type DataGridRendererColumn,
+  type DataGridRendererRow,
+} from "../..";
+import { useTableFeatures } from "./features";
+import { useTableSorting } from "./features/sorting/sorting";
+
+const props = defineProps<{
+  /**
+   * The order of and which columns should be rendered.
+   */
+  columns: (keyof TEntry)[];
+  /**
+   * The data that should be used to fill the table.
+   */
+  data: TEntry[];
+}>();
+
+const withSorting = useTableSorting<TEntry>();
+
+const { watchSources, createRendererRows, createRendererColumns } = useTableFeatures([withSorting]);
+
+const renderCols: Ref<DataGridRendererColumn<TEntry, object>[]> = ref([]);
```

Comment:
```suggestion
const renderColumns: Ref<DataGridRendererColumn<TEntry, object>[]> = ref([]);
```

Isn't it a bad practice to use abbreviations for variable names?